sentry-core/src/main/java/io/sentry/core/SendCachedEventFireAndForgetIntegration.java

```diff
@@ -34,25 +36,21 @@ public final void register(@NotNull IHub hub, @NotNull SentryOptions options) {
       return;
     }
 
-    SendFireAndForget sender = factory.create(hub, options);
+    final SendFireAndForget sender = factory.create(hub, options);
 
     try {
-      ExecutorService es = Executors.newSingleThreadExecutor();
-      es.submit(
-          () -> {
-            try {
-              sender.send();
-              options
-                  .getLogger()
-                  .log(SentryLevel.DEBUG, "Finished processing cached files from %s", cachedDir);
-            } catch (Exception e) {
-              options.getLogger().log(SentryLevel.ERROR, "Failed trying to send cached events.", e);
-            }
-          });
       options
-          .getLogger()
-          .log(SentryLevel.DEBUG, "Scheduled sending cached files from %s", cachedDir);
-      es.shutdown();
+          .getExecutorService()
```

Comment:
H: Was the goal of adding `executorService` to SentryOptions so we can control the `close`? I'm not sure it belongs here, xan we discuss this a bit in more detail before going ahead with this?

So far the only use of it is to be closed during `Hub.close`. If that's the case, could we simply implement `Closeable` on this type and `shutdown` the executor there? And still have a single executor for this type?

Or do we need to share this executor? We could investigate a `SentryExexcutorService` that we'd pass around and could be used by different bits of the SDK?

Reply:
2 things, 1st is indeed to be able to close and not allow a living thread forever if `close` has been called.

my first try was implementing `Closeable`, but we have 3 integrations using the very same executor, it might be the case that you call close for a single integration and this is going to close the executor for the other 2 integrations too, might not be a real case, but still possible.

I decided to expose an executor that can also be used in `AndroidOptionsInitializer`, as we also need to init. the integration for sending NDK events and then, close the executor on `hub.close()`

We need to share the executor to avoid racey conditions and also to minimize a lot of stuff in parallel on the App. start/restart, it's just optimal to execute one after the other using the same executor, this also makes our `RetryingThreadPoolExecutor` to not get a lot of items at the same time and drop a few if >= queue size.

btw the name of the PR is `send cached events to use a single executor`, so obviously the idea is to share an executor for the integrations that send cached events/envelopes, I've updated this PR yesterday so just telling in case you have missed this.

`SentryExexcutorService` is a great idea, I'd go for something like this too, and this goes to SentryOptions, right? as we need to access in other places.
As this is a Draft PR, it's not fully tested and so on, but it did solve the problems I've found, let's discuss.

Reply:
btw this can also help with #361 in the future

Reply:
> my first try was implementing Closeable, but we have 3 integrations using the very same executor, it might be the case that you call close for a single integration and this is going to close the executor for the other 2 integrations too, might not be a real case, but still possible.

Fair but one SentryOptions could be used on a second SentryClient to be bound to the same Hub (i.e to have a second background queue, also not likely a common use case but possible). And the client also `close` too which should flush its events but not necessarily close all other clients/integrations in the app.

Reply:
discussed offline